core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1442,6 +1442,7 @@ private CompiledQuery repairTables(SqlExecutionContext executionContext) throws
 
             try {
                 engine.getWriter(executionContext.getCairoSecurityContext(), tok).close();
```

Comment:
the repair before your change boiled down to opening and closing writer. Writer does repair table when opened. Now that you open writer anyway for upgrade this line is not necessary. May be leave a comment there to say that opening writer attempts to fix things.